### PR TITLE
[74] Expand permission enum to cover the rest of the permissions listed

### PR DESCRIPTION
**Pull request by sam-i-am012**
Addresses #74. All tests pass. 

Not sure if I should include `modify_object` and `read_object` since it's equivalent to `write` and `own`.

### CODE REVIEW

**Comment by korydraughn** [comment]
Please include both forms.

**Comment by sam-i-am012** [comment]
Ok, I have added both forms.

File: src/main/java/org/irods/Util/Permission.java [hunk 1]
```diff
@@ -7,8 +7,16 @@ public enum Permission {
...
-    
+    OWN("own"),
+    DELETE_OBJECT("delete_object"),
+    CREATE_OBJECT("create_object"),
+    DELETE_METADATA("delete_metadata"),
+    MODIFY_METADATA("modify_metadata"),
+    CREATE_METADATA("create_metadata"),
+    READ_METADATA("read_metadata"),
+    MODIFY_OBJECT("modify_object"),
+    READ_OBJECT("read_object");
```
Comment (korydraughn) [inline src/main/java/org/irods/Util/Permission.java:18]: Please group these with their aliases respectively.

**Comment by korydraughn** [comment]
I got a notification about this PR, but there doesn't appear to be anything different?

File: src/main/java/org/irods/Util/Permission.java (hunk 1 above)
Comment (trel) [reply src/main/java/org/irods/Util/Permission.java:18]: please keep `own` on top and `read_object` at the bottom... with the aliases in there/alongside like kory suggested.

that ordering is most easy to parse later when the next reader comes along

Comment (trel) [reply src/main/java/org/irods/Util/Permission.java:18]: sorry, read_object is not at the bottom... please put null at the bottom.   and the rest in decreasing order (they are linear).

https://docs.irods.org/4.3.2/icommands/user/#ichmod

Comment (sam-i-am012) [reply src/main/java/org/irods/Util/Permission.java:18]: ok, so just in the same order as that doc you linked?

Comment (sam-i-am012) [reply src/main/java/org/irods/Util/Permission.java:18]: Done.

**Comment by sam-i-am012** [comment]
Resolved all existing comments. I was rebasing onto main so maybe that's why you got a notification? I'm not sure.

**Review by korydraughn — COMMENTED** [review]
That's probably true.

The changes look good. Please squash to one commit, no pounds.

File: src/main/java/org/irods/Util/Permission.java (hunk 1 above)
Comment (trel) [reply src/main/java/org/irods/Util/Permission.java]: yes, thanks.

**Comment by sam-i-am012** [comment]
Done.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by sam-i-am012** [comment]
Done.

**Comment by korydraughn** [comment]
No conflicts reported by github.

Merging.

**Pull request merged**